packages/lesswrong/server/voteServer.ts

```diff
@@ -267,7 +271,7 @@ const checkRateLimit = async ({ document, collection, voteType, user }: {
   const oneDayAgo = moment().subtract(1, 'days').toDate();
   const votesInLastDay = await Votes.find({
     userId: user._id,
-    authorId: {$ne: user._id}, // Self-votes don't count
+    authorIds: {$ne: user._id}, // Self-votes don't count
```

Comment:
I don't *think* this will work, based on my reading of mongo docs.

Reply:
I tested it out and it seems to work. For instance, in the EA dev db `db.votes.find({ _id: "5be4dc3ce61b803757b01324", authorIds: { $eq: "Yx4AN5v8gJBCtrN67" } })` will return 1 result and `db.votes.find({ _id: "5be4dc3ce61b803757b01324", authorIds: { $ne: "Yx4AN5v8gJBCtrN67" } })` will return 0.

I guess the alternative would be `db.votes.find({ _id: "5be4dc3ce61b803757b01324", $expr: { $in: ["Yx4AN5v8gJBCtrN67", "$authorIds"] } })` and `db.votes.find({ _id: "5be4dc3ce61b803757b01324", $expr: { $not: { $in: ["Yx4AN5v8gJBCtrN67", "$authorIds"] } } })`, but everything I've read (without having actually measured it) suggests this will be much worse for performance.

I've added another couple of tests to cover rate limiting since it's pretty impractical to test manually (it doesn't cover everything, but it does cover this query - we can expand the test in the future if necessary).